src/test/operators/join_index_test.cpp

```diff
@@ -443,7 +459,27 @@ TYPED_TEST(JoinIndexTest, JoinOnDictAndReferenceSegment) {
                          JoinMode::Inner, "resources/test_data/tbl/join_operators/int_inner_join_neq.tbl", 1);
 }
 
-TYPED_TEST(JoinIndexTest, MultiJoinOnReferenceLeft) {
+TYPED_TEST(JoinIndexTest, MultiJoinOnReferenceLeftIndexLeft) {
+  // scan that returns all rows
+  auto scan_a = this->create_table_scan(this->_table_wrapper_f, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_a->execute();
+  auto scan_b = this->create_table_scan(this->_table_wrapper_g, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_b->execute();
+  auto scan_c = this->create_table_scan(this->_table_wrapper_h, ColumnID{0}, PredicateCondition::GreaterThanEquals, 0);
+  scan_c->execute();
+
+  auto join = std::make_shared<JoinIndex>(
+      scan_a, scan_b, JoinMode::Inner, OperatorJoinPredicate{{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals});
+  join->execute();
+
+  // Referencing single chunk guarantee is not given since the left input of the index join is also an index join
+  // and the IndexSide is left. The execution of the join index does not provide single chunk reference guarantee.
```

Comment:
> The execution of the join index

Nitpicky: Do you mean the index join operator? We put the "join" first so that all joins are grouped. When talking about them, please either use, e.g., "JoinHash operator" or "the hash join", but not "the join hash".